pkg/registries/registries.go

```diff
@@ -131,34 +164,70 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		}
 	}
 
-	// insecureRegistries and blockedRegistries are lists of registries; now that mirrors can be configured at a namespace/repo level,
-	// configuration at the namespace/repo level would shadow the registry-level entries; so, propagate the insecure/blocked
+	// Add the blocked registry entries to the registries list so that we can find sub-scopes of insecure registries and set both the
+	// blocked and insecure flags accordingly.
+	// e.g *.blocked.insecure.com is a sub-scope of *.insecure.com and should have both the insecure and blocked options set to true. If
+	// we don't add the blocked registries list to the registries config list before going through the insecure registries we won't be able
+	// to check if *.blocked.insecure.com is a sub-scope of *.insecure.com as it won't exist in the registries config list and will not have
+	// insecure=true, so we need to populate the registries config list with the blocked registries before moving on.
+	for _, scope := range blockedScopes {
+		_ = getRegistryEntry(scope)
+	}
+
+	// any of insecureScopes, blockedScopes, and mirrors, can be configured at a namespace/repo level,
+	// and in V2RegistriesConf, only the most precise match is used; so, propagate the insecure/blocked
 	// flags to the child namespaces as well.
-	for _, insecureReg := range insecureRegistries {
-		reg := getRegistryEntry(insecureReg)
+	for _, insecureScope := range insecureScopes {
+		reg := getRegistryEntry(insecureScope)
 		reg.Insecure = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, insecureReg) {
+			scope := reg.Location
+			// Set scope to Prefix if it exists
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if scopeIsNestedInsideScope(scope, insecureScope) {
 				reg.Insecure = true
 			}
 			for j := range reg.Mirrors {
 				mirror := &reg.Mirrors[j]
-				if scopeMatchesRegistry(mirror.Location, insecureReg) {
+				if scopeIsNestedInsideScope(mirror.Location, insecureScope) {
 					mirror.Insecure = true
 				}
 			}
 		}
 	}
-	for _, blockedReg := range blockedRegistries {
-		reg := getRegistryEntry(blockedReg)
+	for _, blockedScope := range blockedScopes {
+		reg := getRegistryEntry(blockedScope)
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// Set scope to Prefix if it exists
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if scopeIsNestedInsideScope(scope, blockedScope) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// IsValidRegistriesConfScope returns true if scope is a valid registry or wildcard entry such as "*.registry.com"
```

Comment:
```suggestion
// IsValidRegistriesConfScope returns true if scope is a valid scope for the Prefix key in registries.conf
```
or describe all three cases (registry, wildcard, within-registry namespace/repo)